utilities/blob_db/blob_compaction_filter.cc

```diff
@@ -316,12 +387,23 @@ BlobIndexCompactionFilterFactoryGC::CreateCompactionFilter(
 
   assert(blob_db_impl());
 
-  BlobCompactionContext context;
-  BlobCompactionContextGC context_gc;
-  blob_db_impl()->GetCompactionContext(&context, &context_gc);
+  BlobCompactionContext bcontext;
+  BlobCompactionContextGC bcontext_gc;
+  blob_db_impl()->GetCompactionContext(&bcontext, &bcontext_gc);
+
+  auto rv = new BlobIndexCompactionFilterGC(
+      std::move(bcontext), std::move(bcontext_gc), current_time, statistics());
+
+  // compaction_filter takes precedence over compaction_filter_factory if
+  // client specifies both. (refer to include\rocksdb\options.h)
+  if (user_filter_ != nullptr) {
+    rv->SetUserCompactionFilter(user_filter_);
+  } else if (user_filter_factory_ != nullptr) {
+    auto user_filter = user_filter_factory_->CreateCompactionFilter(context);
+    rv->SetUserCompactionFilter(&user_filter);
+  }
```

Comment:
Please look at how the TTLCompactionFilter code is created.  Perhaps we can use a similar model (or common base class)?

Reply:
Good point, add a common class LayeredCompactionFilterBase